docs/code-style-guide.md

```diff
@@ -62,6 +62,44 @@ The vast majority of the things noted in this document will apply to `C` code as
 
 * Functions that take no arguments must be declared as `foo(void)` **not** `foo()`, as the second option allows `foo` to take anything as it's arguments ([reference](https://softwareengineering.stackexchange.com/questions/286490/what-is-the-difference-between-function-and-functionvoid/286494))
 
+* Functions that return values via argument(s) must have all parameters labelled as `[in]`, `[in/out]`, or `[out]` in the javadoc, **in that order**.
```

Comment:
can we avoid `[in/out]` arguments or is that something that is super useful? If possible it would be nice to be strictly `[in]` or `[out]`

Reply:
I assume the `[in/out]` arguments as mentioned were for in-place arguments (i.e arguments you pass in by reference and are modified). So having `[in/out]` seems necessary.

Reply:
Yeah, @kelvinkoon is basically correct here. In general I'm trying to have convention here that somewhat matches what we've been doing, want to avoid trying to constraint what we do per-say, just make sure it's consistent.